### PR TITLE
chore(gateway): update signer hash to signers hash

### PR DESCRIPTION
**Pull request by ahramy**
[AXE-6300](https://axelarnetwork.atlassian.net/browse/AXE-6300)

[AXE-6300]: https://axelarnetwork.atlassian.net/browse/AXE-6300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: evm/deploy-amplifier-gateway.js [hunk 1]
```diff
@@ -260,10 +260,10 @@ async function deploy(config, chain, options) {
                 error = true;
             }
 
-            if (signerHashByEpoch !== signerHash) {
+            if (signersHashByEpoch !== signersHash) {
                 printError(`ERROR: Signer hash mismatch for signer set ${i + 1}`);
```
Comment (milapsheth) [inline evm/deploy-amplifier-gateway.js:264]: you can release a snapshot to test that this will work correctly before a proper release

**Review by milapsheth — APPROVED** [review]

File: evm/deploy-amplifier-gateway.js (hunk 1 above)
Comment (ahramy) [reply evm/deploy-amplifier-gateway.js:264]: done

File: evm/gateway.js [hunk 2]
```diff
@@ -81,7 +81,7 @@ async function processCommand(config, chain, options) {
     printInfo('Contract name', contractName);
     printInfo('Contract address', gatewayAddress);
 
-    const gateway = new Contract(gatewayAddress, IGateway.abi, wallet);
+    const gateway = new Contract(gatewayAddress, AxelarGateway.abi, wallet);
```
Comment (milapsheth) [inline evm/gateway.js:84]: depending on gateway type (there's a connectionType field in config), can you initialize it to amplifier vs old gateway?

**Review by milapsheth — APPROVED** [review]

File: evm/gateway.js (hunk 2 above)
Comment (ahramy) [reply evm/gateway.js]: done

File: evm/gateway.js [hunk 3]
```diff
@@ -28,7 +28,8 @@ const {
 const { addBaseOptions } = require('./cli-utils');
 const { getWallet, signTransaction } = require('./sign-utils');
 
-const IGateway = require('@axelar-network/axelar-gmp-sdk-solidity/interfaces/IAxelarGateway.json');
+const AmplifierGateway = require('@axelar-network/axelar-gmp-sdk-solidity/interfaces/IAxelarGateway.json');
```
Comment (milapsheth) [inline evm/gateway.js:31]: this interface only supports messaging. you have to use the interface corresponding to the actual amplifier gateway

**Review by milapsheth — APPROVED** [review]

File: evm/gateway.js (hunk 3 above)
Comment (ahramy) [reply evm/gateway.js]: aha, thanks for pointing it out. updated.

**Pull request merged**